pkg/kvstore/store/store.go

```diff
@@ -363,6 +376,10 @@ func (s *SharedStore) SharedKeysMap() map[string]Key {
 
 // UpdateLocalKey adds a key to be synchronized with the kvstore
 func (s *SharedStore) UpdateLocalKey(key LocalKey) {
+	if s.conf.ReadOnly {
+		s.getLogger().Fatal("Bug: Local key updated via read-only store")
```

Comment:
Nit:

In the codebase, I've seen this used as "BUG". It would be good to stay consistent.